src/hooks/useStory.ts

```diff
@@ -1,34 +1,112 @@
+import { useNavigate, useParams } from 'react-router-dom';
 
-import { deleteStory, postStory } from '../apis/story';
+import { deleteStory, getStoryDetail, postStory } from '../apis/story';
+import { isBlankString } from '../utils/validations';
+import { putStory } from './../apis/story';
 import { ERROR_MESSAGES } from './../constants/errorMessages';
 
+export const useFetchStory = () => {
+  const [story, setStory] = useState({
+    likes: [],
+    comments: [],
+    _id: '',
+    image: '',
+    imagePublicId: '',
+    title: '',
+    channel: {},
+    author: { image: '', fullName: '', _id: '', role: '' },
+    createdAt: '',
+    updatedAt: '',
+  });
+  const [isLoading, setIsLoading] = useState(false);
+  const [isNew, setIsNew] = useState(false);
+  const { storyId } = useParams();
+
+  useEffect(() => {
+    const fetchStory = async () => {
+      setIsLoading(true);
+      try {
+        if (!storyId) return Error('잘못된 스토리 접근(storyId)'); // TODO: 404 페이지로 리다이렉트
+
+        if (storyId === 'new') {
+          setIsNew(true);
+        } else {
+          const fetchedStories = await getStoryDetail(storyId);
+          setStory(fetchedStories);
+        }
+      } catch (error) {
+        console.error(error);
+        alert(ERROR_MESSAGES.INVOKED_ERROR_GETTING_STORY);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchStory();
+  }, []);
+
+  const fetchComment = async () => {
+    try {
+      if (storyId) {
+        const fetchedStories = await getStoryDetail(storyId);
+        setStory({ ...story, comments: fetchedStories.comments });
+      }
+    } catch (error) {
+      console.error(error);
+      alert(ERROR_MESSAGES.INVOKED_ERROR_GETTING_COMMENT);
+    }
+  };
+
+  return { story, isNew, isLoading, fetchComment };
+};
+
 const getDateInfo = (date: Dayjs) => ({
   year: date.get('year'),
   month: date.get('month') + 1,
   date: date.get('date'),
 });
+interface StoryInfo {
+  title: string;
+  date: {
+    year: number;
+    month: number;
+    date: number;
+  };
+  imageURL?: string;
+  content: string;
+}
 
-const isBlank = (string: string) => {
-  return string.trim().length === 0;
-};
+interface Params {
+  initialValues?: StoryInfo;
+}
 
-export const useStoryForm = () => {
-  const today = dayjs(new Date());
+export const useStoryForm = ({ initialValues }: Params) => {
   const channelId = '63b6822ade9d2a22cc1d45c3';
-  const [values, setValues] = useState({
-    title: '',
-    date: getDateInfo(today),
-    content: '',
+  const today = dayjs(new Date());
+
+  const [values, setValues] = useState(
+    initialValues || {
+      title: '',
+      date: getDateInfo(today),
+      content: '',
+    }
+  );
+  const [date, setDate] = useState<Dayjs | null>(() => {
+    if (initialValues && Object.keys(initialValues.date).length) {
+      const { year, month, date } = initialValues.date;
+      return dayjs(new Date(year, month - 1, date));
+    } else {
+      return today;
+    }
   });
```

Comment:
```suggestion
const [date, setDate] = useState<Dayjs | null>(() => {
  if (initialValues && Object.keys(initialValues.date).length) {
    const { year, month, date } = initialValues.date;
    return dayjs(new Date(year, month - 1, date));
  } 
  
  return today;
});
```

요거는 어떨까 생각이 드네요. 개인 취향!

Reply:
너무 좋습니다 갓승준님